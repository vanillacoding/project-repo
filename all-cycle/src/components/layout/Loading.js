import styled, { keyframes } from "styled-components";
import { faRecycle } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const Container = styled.div`
  width: 100vw;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 1em;
  color: ${(props) => props.theme.skyblue.color};
  background-color: ${(props) => props.theme.white.color};
  z-index: 10;
`;

const rotate = keyframes`
  50% {
    transform: rotateZ(360deg);
  }
  100% {
    transform: rotateZ(720deg);
  }
`;

const RecycleMark = styled.div`
  color: ${(props) => props.theme.skyblue.color};
  animation: ${rotate} 3s infinite cubic-bezier(0.3, 0, 0.8, 1);
  z-index: 10;
  margin-top: 1em;
`;

function Loading() {
  return (
    <Container>
      <div>ALL CYCLE</div>
      <RecycleMark>
        <FontAwesomeIcon icon={faRecycle} size="5x" />
      </RecycleMark>
    </Container>
  );
}

export default Loading;
