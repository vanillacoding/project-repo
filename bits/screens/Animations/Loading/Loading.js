import React from 'react';
import LottieView from 'lottie-react-native';
import { NUMBERS, COLORS } from '../../../constants/index';

const Loading = () => {
  return (
    <>
      <LottieView
        source={require('../../../assets/animation/loadingAnimation.json')}
        autoPlay
        loop
        speed={NUMBERS.TWICE_ANIMATION_SPEED}
        style={{ backgroundColor: COLORS.LOADING_ANIMATION_BACKGROUND }}
      />
    </>
  );
};

export default Loading;
