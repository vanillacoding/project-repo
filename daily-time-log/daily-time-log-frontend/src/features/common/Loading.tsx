import React from "react";
import styled from "styled-components";

function Loading() {
  return (
    <LoadingStyled>
      <Loader />
    </LoadingStyled>
  );
}

const Loader = styled.div`
  border: 16px solid #f3f3f3;
  border-top: 16px solid ${({ theme }) => theme.palette.blue};
  border-radius: 50%;
  width: 120px;
  height: 120px;
  animation: spin 1.5s linear infinite;

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

const LoadingStyled = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
`;

export default Loading;
