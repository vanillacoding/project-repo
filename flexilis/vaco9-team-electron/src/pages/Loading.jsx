import React, { useEffect } from 'react';
import styled from 'styled-components';
import { ipcRenderer } from 'electron';
import { useDispatch } from 'react-redux';
import { fetchLogin } from '../reducers/login';
import Spinner from '../components/Spinner.jsx';
import { color } from '../css/color';

const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
  width: 100vw;
  background-color: ${color.MAIN};
`;

export default function Loading() {
  const dispatch = useDispatch();

  useEffect(async () => {
    ipcRenderer.on('loginDataExist', async (event, data) => {
      const userData = {
        email: data.account,
        password: data.password,
      };

      dispatch(fetchLogin(userData));
    });

    ipcRenderer.on('loginDataDoesNotExist', (event, data) => {
      window.location.hash = '#/login';
    });
  });

  return (
    <Container>
      <Spinner
        color="white"
        loading={true}
        size={40}
      />
    </Container>
  );
}
