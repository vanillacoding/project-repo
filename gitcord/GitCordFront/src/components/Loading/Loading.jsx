import React, { useEffect, useState } from "react";
import styled from "styled-components";

import ModalBackground from "../publicComponents/ModalBackground/ModalBackground";

const Loader = styled.div`
  position: relative;
  height: 5rem;
  width: 9rem;
  animation-name: rotate;
  animation-duration: 3s;
  animation-delay: 1s;
  animation-iteration-count: infinite;

@keyframes rotate {
  from {
    transform: rotate(0deg);
  }
  80% {
    transform: rotate(720deg);
  }
  to {
    transform: rotate(720deg);
  }
}

.l-small {
  position: absolute;
  height: 1rem;
  width: 1rem;
  background: #eee;
  left: 0;
  top: 2rem;
  animation-name: left-small;
  animation-duration: 3s;
  animation-delay: 1s;
  animation-iteration-count: infinite;
}

@keyframes left-small {
  from {
    transform: translate(0rem, 0rem);
  }
  20% {
    transform: translate(-2rem, -1rem);
  }
  60% {
    transform: translate(3rem, -3rem);
  }
  70% {
    transform: translate(-2rem, -1rem);
  }
  80% {
    transform: translate(0rem, 0rem);
  }
  to {
    transform: translate(0rem, 0rem);
  }
}

.l-big {
  position: absolute;
  height: 2rem;
  width: 2rem;
  background: #eee;
  left: 1rem;
  top: 1.5rem;
  animation-name: left-big;
  animation-duration: 3s;
  animation-delay: 1s;
  animation-iteration-count: infinite;
}

@keyframes left-big {
  from {
    transform: translateX(0rem);
  }
  20% {
    transform: translateX(-1rem);
  }
  70% {
    transform: translateX(-1rem);
  }
  80% {
    transform: translateX(0rem);
  }
  to {
    transform: translateX(0rem);
  }
}

.face {
  position: absolute;
  height: 3rem;
  width: 3rem;
  background: #eee;
  left: 3rem;
  top: 1rem;
  border-radius: 20%;
}

.face::before {
  position: absolute;
  content: "";
  height: 0.8rem;
  width: 0.75rem;
  background: #262a39;
  border-radius: 50%;
  top: 1rem;
  left: 0.5rem;
}

.face::after {
  position: absolute;
  content: "";
  height: 0.8rem;
  width: 0.75rem;
  background: #262a39;
  border-radius: 50%;
  top: 1rem;
  right: 0.5rem;
}

.r-big {
  position: absolute;
  height: 2rem;
  width: 2rem;
  background: #eee;
  right: 1rem;
  top: 1.5rem;
  animation-name: right-big;
  animation-duration: 3s;
  animation-delay: 1s;
  animation-iteration-count: infinite;
}

@keyframes right-big {
  from {
    transform: translateX(0rem);
  }
  20% {
    transform: translateX(1rem);
  }
  70% {
    transform: translateX(1rem);
  }
  80% {
    transform: translateX(0rem);
  }
  to {
    transform: translateX(0rem);
  }
}

.r-small {
  position: absolute;
  height: 1rem;
  width: 1rem;
  background: #eee;
  right: 0;
  top: 2rem;
  animation-name: right-small;
  animation-duration: 3s;
  animation-delay: 1s;
  animation-iteration-count: infinite;
}

@keyframes right-small {
  from {
    transform: translate(0rem, 0rem);
  }
  20% {
    transform: translate(2rem, -1rem);
  }
  30% {
    transform: translate(1rem, 3rem);
  }
  40% {
    transform: translate(-2rem, 3rem);
  }
  50% {
    transform: translate(-2rem, 3rem);
  }
  60% {
    transform: translate(1rem, 3rem);
  }
  70% {
    transform: translate(2rem, -1rem);
  }
  80% {
    transform: translate(0rem, 0rem);
  }
  to {
    transform: translate(0rem, 0rem);
  }
}
`;

const LoadingContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  width: 600px;
  height: 250px;

  .text {
    font-size: 50px;
    font-weight: bold;
    color: white;
  }
`;

function Loading() {
  const [loadingText, setLoadingText] = useState("Loading");

  useEffect(() => {
    const loadingCount = setTimeout(() => {
      if (loadingText === "Loading...") {
        setLoadingText("Loading");
      } else {
        setLoadingText(loadingText + ".");
      }
    }, 500);

    return () => clearTimeout(loadingCount);
  }, [loadingText]);

  return (
    <ModalBackground>
      <LoadingContainer>
        <Loader>
          <span className="l-small"></span>
          <span className="l-big"></span>
          <span className="face"></span>
          <span className="r-big"></span>
          <span className="r-small"></span>
        </Loader>
        <div className="text">{loadingText}</div>
      </LoadingContainer>
    </ModalBackground>
  );
}

export default Loading;
