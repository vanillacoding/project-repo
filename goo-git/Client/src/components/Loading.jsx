import React, { useState, useEffect } from 'react';
import { IslandWrapper } from '../styledComponents/IslandWrapper.styled';

export default function Loading({ text }) {
  const [content, setContent] = useState(text);

  useEffect(() => {
    const interval = window.setInterval(() => {
      content === `${text}...`
        ? setContent(text)
        : setContent(`${content}.`);
    }, 300);

    return () => {
      window.clearInterval(interval);
    };
  });

  return (
    <IslandWrapper>
      <h1>{content}</h1>
    </IslandWrapper>
  );
}
