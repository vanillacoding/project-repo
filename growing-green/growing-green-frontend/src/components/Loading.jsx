import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Loading = ({ size = '80px', text = '로딩중...' }) => {
  return (
    <Wrapper>
      <LoadingSpinner size={size} data-testid="loading-spinner" />
      <p>{text}</p>
    </Wrapper>
  );
};

Loading.propTypes = {
  size: PropTypes.string,
  text: PropTypes.string,
};

const Wrapper = styled.div`
  display: flex;
  width: 100%;
  height: 100%;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const LoadingSpinner = styled.div`
  width: ${({ size }) => size};
  height: ${({ size }) => size};
  border-radius: 50%;
  border: 6px solid #999;
  box-sizing: border-box;
  -webkit-animation: rsm-sweep 1s linear alternate infinite,
    rsm-rotate 0.8s linear infinite;
  animation: rsm-sweep 1s linear alternate infinite,
    rsm-rotate 0.8s linear infinite;

  @keyframes rsm-rotate {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }

  @keyframes rsm-sweep {
    0% {
      -webkit-clip-path: polygon(
        0% 0%,
        0% 0%,
        0% 0%,
        50% 50%,
        0% 0%,
        0% 0%,
        0% 0%
      );
      clip-path: polygon(0% 0%, 0% 0%, 0% 0%, 50% 50%, 0% 0%, 0% 0%, 0% 0%);
    }
    50% {
      clip-path: polygon(
        0% 0%,
        0% 100%,
        0% 100%,
        50% 50%,
        100% 0%,
        100% 0%,
        0% 0%
      );
    }
    100% {
      clip-path: polygon(
        0% 0%,
        0% 100%,
        100% 100%,
        50% 50%,
        100% 100%,
        100% 0%,
        0% 0%
      );
    }
  }
`;

export default Loading;
