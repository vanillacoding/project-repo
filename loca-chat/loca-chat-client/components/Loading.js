import React, { useEffect, useState } from 'react';
import { View, Text, StyleSheet } from 'react-native';

export default function Loading() {
  const [dot, setDot] = useState('');

  useEffect(() => {
    const interval = setInterval(() => {
      if (dot.length === 3) {
        setDot('');
      } else {
        setDot((dot) => dot + '.');
      }
    }, 100);

    return () => clearInterval(interval);
  });

  return (
    <View style={styles.container}>
      <Text style={styles.text}>{dot}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  text: {
    fontSize: 100
  }
});
