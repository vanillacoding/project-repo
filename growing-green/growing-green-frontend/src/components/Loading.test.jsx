import React from 'react';
import 'jest-styled-components';
import Loading from './Loading';
import { render } from '../test/test-utils';

describe('<Loading>', () => {
  it('matches snapshot', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
  
  it('should have given style elements', () => {
    const sampleSize = '100px';
    const sampleText = 'sample';
    const { getByTestId } = render(
      <Loading size={sampleSize} text={sampleText} />,
    );
    expect(getByTestId('loading-spinner')).toHaveStyleRule('width', sampleSize);
    expect(getByTestId('loading-spinner')).toHaveStyleRule(
      'height',
      sampleSize,
    );
  });
});
