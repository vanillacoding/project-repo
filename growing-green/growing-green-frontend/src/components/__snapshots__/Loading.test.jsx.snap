// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading> matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 80px;
  height: 80px;
  border-radius: 50%;
  border: 6px solid #999;
  box-sizing: border-box;
  -webkit-animation: rsm-sweep 1s linear alternate infinite, rsm-rotate 0.8s linear infinite;
  -webkit-animation: rsm-sweep 1s linear alternate infinite, rsm-rotate 0.8s linear infinite;
  animation: rsm-sweep 1s linear alternate infinite, rsm-rotate 0.8s linear infinite;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="loading-spinner"
    />
    <p>
      로딩중...
    </p>
  </div>
</div>
`;
